neuralpp/test/quick_tests/graphical_model/representation/belief_propagation_test.py

```diff
@@ -0,0 +1,37 @@
+from neuralpp.experiments.experimental_inference.belief_propagation import BeliefPropagation
+from neuralpp.inference.graphical_model.representation.factor.pytorch_table_factor import (
+    PyTorchTableFactor,
+)
+from neuralpp.inference.graphical_model.variable.integer_variable import IntegerVariable
+
+
+def test_belief_propagation():
+    cloudy = [0.2, 0.4, 0.4]
+    sprinkler = [0.6, 0.4]
+    rain = [
+        [0.2, 0.3, 0.5],
+        [0.3, 0.4, 0.3],
+        [0.4, 0.2, 0.4]
+    ]
+
+    def wet_grass(wetness: int, rain_level: int, sprinkler_on: int):
+        return 1.0 if (rain_level + sprinkler_on == wetness) else 0.0
+
+    for log_space in {False, True}:
+        c = IntegerVariable("c", 3)
+        r = IntegerVariable("r", 3)
+        s = IntegerVariable("s", 2)
+        w = IntegerVariable("w", 4)
+
+        model = [
+            PyTorchTableFactor([c], cloudy, log_space=log_space),
+            PyTorchTableFactor([c, r], rain, log_space=log_space),
+            PyTorchTableFactor([s], sprinkler, log_space=log_space),
+            PyTorchTableFactor.from_function(
+                [w, r, s], wet_grass, log_space=log_space
+            ),
+        ]
+
+        query = w
+        print(BeliefPropagation().run(query, model)) # [0.192, 0.308, 0.348, 0.152]
```

Comment:
Note that, in principle, you might not get a normalized distribution here (for example when you have observations). So you probably want to use `normalize()` at the end of `BeliefPropagation.run`.

BTW, please add a test in which we observe `cloudy` to be `2` (the maximum value). To incorporate observations, condition all factors on dict `{c: 2}`. Do we see an increase in the probability of `wet_grass`? Remember that you will need to include a normalization in the inference algorithm in  order to obtain a normalized distribution that makes more sense.